akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MethodDirectivesSpec.scala

```diff
@@ -28,21 +29,60 @@ class MethodDirectivesSpec extends RoutingSpec {
   }
 
   "head" should {
-    val headRoute = head {
-      complete(HttpEntity.Default(
+    val emptyComplete = complete(
+      HttpEntity.Default(
         ContentTypes.`application/octet-stream`,
         12345L,
         Source.empty
-      ))
+      )
+    )
+
+    val headRoute = head {
+      emptyComplete
+    }
+
+    val getRoute = get {
+      emptyComplete
+    }
+
+    def responseEntityEmpty() = {
+      val lengthF = response._3.dataBytes.runFold(0)((c, _) ⇒ c + 1)
+      val length = Await.result(lengthF, Duration(100, "millis"))
+      length shouldEqual 0
     }
 
-    "allow manual complete" in {
+    "allow manual complete if transparent head is enabled" in {
+      implicit val settings = RoutingSettings.default.withTransparentHeadRequests(true)
+
       Head() ~> headRoute ~> check {
         status shouldEqual StatusCodes.OK
+        responseEntityEmpty()
+      }
+    }
+
+    "allow manual complete if transparent head is disabled" in {
+      implicit val settings = RoutingSettings.default.withTransparentHeadRequests(false)
 
-        val lengthF = response._3.dataBytes.runFold(0)((c, _) ⇒ c + 1)
-        val length = Await.result(lengthF, Duration(100, "millis"))
-        length shouldEqual 0
+      Head() ~> headRoute ~> check {
+        status shouldEqual StatusCodes.OK
+        responseEntityEmpty()
+      }
+    }
```

Comment:
do we really need these `Head() ~> headRoute` tests? maybe add a `checkHead(route: Route, transparentHeadRequests: Boolean)` helper?